repos/emacs/update-unstable.py

```diff
@@ -0,0 +1,72 @@
+from packaging.version import Version, InvalidVersion
+import subprocess
+import json
+
+# (Ab)use Python's PEP-440 version parsing for sorting Emacs versions.
+# Emacs versions aren't exactly PEP-440, but `git ls-remote`'s sort function does not support -rc
+# suffixes and neither does GNU sort.
+# Meaning that older -rc releases may be prefered to a later stable release.
+
+TAG_PREAMBLE = "refs/tags/emacs-"
+
+
+def main():
+    proc = subprocess.run(
+        [
+            "git",
+            "ls-remote",
+            "--tags",
+            "--refs",
+            "https://git.savannah.gnu.org/git/emacs.git",
+            "emacs-[1-9]*",
+        ],
+        stdout=subprocess.PIPE,
+        check=True,
+    )
+
+    tags: list[str] = []
+    for line in proc.stdout.decode().splitlines():
+        _commit, ref = line.split("\t")
+        if not ref.startswith(TAG_PREAMBLE):
+            continue
+
+        tag = ref[len(TAG_PREAMBLE) :]
+
+        # Skip unparseable versions
+        try:
+            Version(tag)
+        except InvalidVersion:
+            pass
```

Comment:
Do these tags really exist?  If so, should they be skipped?

Reply:
There is this one: `emacs-24.5-rc3-fixed` that fails to parse.
Apart from that one everything parses just fine.

Reply:
Ah, it is `24.5-rc3-fixed`.  Hopefully there will be no similar tags in the future.

Reply:
Yeah. Maybe it would be better to have a list of known "bad" versions to skip instead so we would catch any newly introduced strangeness?

Reply:
Sounds good.